authentication/auth.go

```diff
@@ -0,0 +1,162 @@
+package authentication
+
+import (
+	"fmt"
+	"log"
+	"time"
+	"errors"
+	"regexp"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/pem"
+	"io/ioutil"
+	"net/url"
+
+	"github.com/satori/go.uuid"
+
+	"gopkg.in/square/go-jose.v2"
+	"gopkg.in/square/go-jose.v2/jwt"
+
+	"github.com/collisdigital/go-launch-a-survey/settings"
+)
+
+func loadEncryptionKey() (key *rsa.PublicKey, error error) {
+
+	encryptionKeyPath := settings.GetSetting("JWT_ENCRYPTION_KEY_PATH")
+	if keyData, err := ioutil.ReadFile(encryptionKeyPath); err == nil {
+		block, _ := pem.Decode(keyData)
+		pub, err := x509.ParsePKIXPublicKey(block.Bytes)
+		if err == nil {
+			return pub.(*rsa.PublicKey), nil
+		}
+	}
+	return nil, errors.New("Failed to load encryption key")
```

Comment:
strongly typed errors are better (allowing things outside the package to check - e.g. `EOF` from the `io` package)

```
var ErrLoadingEncryptionKey = errors.New("Failed to load encryption key")
func foo() error {
    return ErrLoadingEncryptionKey
}

...

err := foo()
if err == ErrLoadingEncryptionKey {
    // handle this particular error type
}
```